### PR TITLE
Execute command on device.

### PR DESCRIPTION
**Pull request by zhou9584**

### CODE REVIEW

**Review by olivershen-wow — DISMISSED** [review]

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/ActionExecutor.java [hunk 1]
```diff
@@ -28,7 +28,8 @@ public class ActionExecutor {
     /**
      * the implementation of supported actions should not be overload
      */
-    private Set<String> actionTypes = Set.of("setProperty", "setDefaultLauncher", "backToHome", "changeGlobalSetting", "changeSystemSetting");
+    private Set<String> actionTypes = Set.of("setProperty", "setDefaultLauncher", "backToHome", "changeGlobalSetting",
```
Comment (hydraxman) [inline agent/src/main/java/com/microsoft/hydralab/agent/runner/ActionExecutor.java:31]: Suggest we migrate this to application.yml and make it configurable at the agent level, and the agent admin can specify this themselves.

We may do this in another PR.

Comment (zhou9584) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/ActionExecutor.java:31]: Will do this work in another PR. If use application.yml to config, we need to add ActionExecutor to bean factory.

File: agent/src/main/java/com/microsoft/hydralab/agent/AgentApplication.java [hunk 2]
```diff
@@ -2,19 +2,24 @@
...
+import com.microsoft.hydralab.common.entity.agent.DeviceCommand;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.autoconfigure.domain.EntityScan;
 import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.PropertySource;
```
Comment (hydraxman) [inline agent/src/main/java/com/microsoft/hydralab/agent/AgentApplication.java:14]: abc

File: agent/src/main/resources/application-test.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+          test command 1
+      - type: ADBShell
+        when: setUp
+        matcher: '.*test.*'
+        inline: |
+          test command 2
+          test command 3
+      - type: ADBShell
+        when: tearDown
+        matcher: 'com.microsoft.test'
```
Comment (hydraxman) [inline agent/src/main/resources/application-test.yml:17]: If it's a matcher for app name, do you think we can name it as packageNameMatcher?

Comment (hydraxman) [reply agent/src/main/resources/application-test.yml:17]: Or matcher:
          package-name: 'com.microsoft.test'

Comment (zhou9584) [reply agent/src/main/resources/application-test.yml:17]: Actually, the matcher is used to match the suite Class name. I will rename it as suite-class-matcher and add a comment: If the suite-class doesn't be provided, the value of suite-class would be the package name of app.

File: common/src/main/java/com/microsoft/hydralab/common/entity/agent/DeviceCommand.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+package com.microsoft.hydralab.common.entity.agent;
+
+import lombok.Data;
+
+/**
+ * @author zhoule
+ * @date 01/31/2023
+ */
+@Data
+public class DeviceCommand {
```
Comment (hydraxman) [inline common/src/main/java/com/microsoft/hydralab/common/entity/agent/DeviceCommand.java:12]: Rename this to DeviceScriptCommand to map the name defined in Yaml, and move this to agent project, under the same package with CommandActionLoader, so these will only be accessible in the agent project.

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/CommandActionLoader.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * @author zhoule
+ * @date 01/31/2023
+ */
+@Service
+public class CommandActionLoader {
```
Comment (hydraxman) [inline agent/src/main/java/com/microsoft/hydralab/agent/runner/CommandActionLoader.java:21]: Rename this to DeviceScriptCommandLoader, and move it to agent project.

File: agent/src/main/java/com/microsoft/hydralab/agent/AgentApplication.java [hunk 6]
```diff
@@ -50,4 +55,10 @@ private static boolean decideHeadlessFromArguments(String[] args) {
         }
         return headless;
     }
+
+    @ConfigurationProperties(prefix = "app.device-script.commands")
+    @Bean(name = "DeviceCommandProperty")
+    public List<DeviceCommand> DeviceCommandProperty() {
```
Comment (hydraxman) [inline agent/src/main/java/com/microsoft/hydralab/agent/AgentApplication.java:61]: Move this to a config with test init.

**Review by hydraxman — APPROVED** [review]
